### DISCUSSION TITLE
HomeConnect Oven Preheat Finished not available in HomeKit?

### DISCUSSION THREAD

**Discussion by McJohnPi** [thoukydides/homebridge-homeconnect#159] [General]
Thanks for this nice plugin. Family is happy with you! 
Everything is working great (e.g., oven open and closed are available in HomeKit) but the "Regular Preheat Finished" and "Preheat Finished" aren't visible into HomeKit. (we can see the big tiles into the "Homebridge Accessories" window). And we saw also an other? "accessory in the log:
[8/12/2023, 5:37:07 PM] [HomeConnect] [Oven] Event **Fast** Preheat Finished
Or is Fast Preheat the same as "Preheat Finished"?
Anyway; they aren't visible in the HomeKit accessories window neither into an HomeKit automation or Scene flow.
Did we do something wrong?
Thanks for your support.
John

**Comment by thoukydides** [comment]
Oven appliances expose five events to HoneKit as [Stateless Programmable Switch](https://github.com/thoukydides/homebridge-homeconnect/wiki/HomeKit-Mapping#stateless-programmable-switch-services-for-events) services. The Apple Home apps shows these as buttons with Single Press triggers that can be used for automations:

![image](https://github.com/thoukydides/homebridge-homeconnect/assets/4775242/ee14f1ba-bc65-4ff7-8115-a3dce8b57bfe)

Unfortunately Apple’s app only shows numeric labels, so you need to know that these correspond to the following events:
1. Program Finished
2. Program Aborted
3. Alarm Clock Finished
4. Fast Preheat Finished
5. Regular Preheat Finished

Some other HomeKit apps show more descriptive names, e.g. Eve:

![image](https://github.com/thoukydides/homebridge-homeconnect/assets/4775242/e1c44435-a922-40f0-a45f-c4dc275b00ab)

As you can see, two different preheat finished events are available. From the [Home Connect documentation](https://api-docs.home-connect.com/events/#program-progress-events):

* Fast Preheat Finished (`Cooking.Oven.Event.PreheatFinished`): This event is triggered when program option FastPreheat is active and the preheating phase is finished.
* Regular Preheat Finished (`Cooking.Oven.Event.RegularPreheatFinished`): This event is triggered when program option FastPreheat is not active and the preheating phase is finished. Please note that the option is not supported by all ovens.

**Comment by thoukydides** [comment]
This was discussed recently under issue #153.

**Comment by McJohnPi** [comment]
Wow! You are fast. Thank you so much for the great support and the hints.
Found the Preheats in the Eve app (Used that also for the dryer.)

Case closed.
